modules/Bio/EnsEMBL/Variation/TranscriptVariationAllele.pm

```diff
@@ -96,6 +96,435 @@ sub new_fast {
+  
+  my $unshifted_allele_string = $self->allele_string;
+  my @allele_string = split('/', $unshifted_allele_string);
+  my $hgvs_output_string = $allele_string[1];
+  
+  
+  ## isolate correct sequence to attempt to shift
+  my $seq_to_check;
+  my $type;
+  if ($var_class eq 'indel')
+  {
+    return $self;
+  }
+  elsif ($var_class eq 'deletion')
+  {
+    $seq_to_check = $allele_string[0];
+    $type = 'del';
+  }
+  elsif ($var_class eq 'insertion') 
+  {
+    $seq_to_check = $allele_string[1];
+    $type = 'ins';
+  }
+  
+  my $shift_length;
+  my $strand = $tr->strand;
+  
+  ## Fix this - I'm sure Perl has a smoother way of doing this
+  my ($a, $b, $c, $d, $e);
+  
+  ## Actually performs the shift, and provides raw data in order to create shifting hash
+  ($shift_length, $seq_to_check, $hgvs_output_string, $start, $end) = $self->perform_shift($seq_to_check, $post_seq, $pre_seq, $start, $end, $hgvs_output_string, (-1 * ($strand -1))/2); 
+  
+  ## Creates shift_hash to attach to VF and TVA objects for 
+  $self->create_shift_hash($seq_to_check, $post_seq, $pre_seq, $start, $end, $hgvs_output_string, $type, $shift_length, $strand, 0);
+
+  return $self;
+}
+
+
+sub check_tva_shifting_hashes{
+  my $self = shift;
+  my $preshifted_hashs = $self->base_variation_feature->{tva_shift_hashs};  
+  my @shift_hashes = grep {$_->{unshifted_allele_string} eq $self->allele_string  && $self->transcript->strand eq $_->{strand}} @{$preshifted_hashs} if defined($preshifted_hashs);
+  if (scalar(@shift_hashes) == 1)
+  {
+    return $shift_hashes[0];
+  }
+  return undef;
+}
+
+=head2 perform_shift
+
+  Description: Performs the shifting calculation
+               Requires flanking sequences, allele string, strand and feature location
+  Returntype : 5 scalars
+  Status     : Stable
+
+=cut
+
+sub perform_shift
+{
+  ## Performs the shifting calculation
+  my ($self, $seq_to_check, $post_seq, $pre_seq, $var_start, $var_end, $hgvs_output_string, $reverse) = @_;
+  ## get length of pattern to check 
+  $reverse = 0 if ($reverse eq "");
+  my $indel_length = (length $seq_to_check);
+  my $shift_length = 0;
+  
+  ## Sets up the for loop, ensuring that the correct bases are compared depending on the strand
+  my $loop_limiter = $reverse ? (length($pre_seq) - $indel_length) + 1 : (length($post_seq) - $indel_length);
+  $loop_limiter = length($post_seq) if $loop_limiter < 0;
+  for (my $n = $reverse; $n <= $loop_limiter; $n++ ){
+    ## check each position in deletion/ following seq for match
+    my $check_next_del  = $reverse ? substr( $seq_to_check, length($seq_to_check) -1, 1) : substr( $seq_to_check, 0, 1);
+    my $check_next_pre = $reverse ? substr( $pre_seq, length($pre_seq) - $n, 1) : substr( $post_seq, $n, 1);;
+    my $hgvs_next_del  = $reverse ? substr( $hgvs_output_string, length($hgvs_output_string) -1, 1) : substr( $hgvs_output_string, 0, 1);;
+    
+    if($check_next_del eq $check_next_pre){
+      ## move position of deletion along
```

Comment:
There are 5 different "if reverse's" in here which could be tidied for readabily, including a couple of
$a = $b if $reverse
$a = $a. $c if $reverse